Firestore/Example/Tests/API/FIRQuerySnapshotTests.m

```diff
@@ -0,0 +1,116 @@
+                                                           oldDocuments:oldDocuments
+                                                        documentChanges:documentChanges
+                                                              fromCache:YES
+                                                       hasPendingWrites:NO
+                                                       syncStateChanged:YES];
+  XCTAssertEqualObjects([FIRQuerySnapshot snapshotWithFirestore:firestore
+                                                  originalQuery:queryFoo
+                                                       snapshot:snapshotFoo
+                                                       metadata:metadataFoo],
+                        [FIRQuerySnapshot snapshotWithFirestore:firestore
+                                                  originalQuery:queryFoo
+                                                       snapshot:snapshotFoo
+                                                       metadata:metadataFoo]);
+  XCTAssertNotEqualObjects([FIRQuerySnapshot snapshotWithFirestore:firestore
+                                                     originalQuery:queryFoo
+                                                          snapshot:snapshotFoo
+                                                          metadata:metadataFoo],
+                           [FIRQuerySnapshot snapshotWithFirestore:firestore
+                                                     originalQuery:queryBar
+                                                          snapshot:snapshotFoo
+                                                          metadata:metadataFoo]);
+  XCTAssertNotEqualObjects([FIRQuerySnapshot snapshotWithFirestore:firestore
+                                                     originalQuery:queryFoo
+                                                          snapshot:snapshotFoo
+                                                          metadata:metadataFoo],
+                           [FIRQuerySnapshot snapshotWithFirestore:firestore
+                                                     originalQuery:queryFoo
+                                                          snapshot:snapshotBar
+                                                          metadata:metadataFoo]);
+  XCTAssertNotEqualObjects([FIRQuerySnapshot snapshotWithFirestore:firestore
+                                                     originalQuery:queryFoo
+                                                          snapshot:snapshotFoo
+                                                          metadata:metadataFoo],
+                           [FIRQuerySnapshot snapshotWithFirestore:firestore
+                                                     originalQuery:queryFoo
+                                                          snapshot:snapshotFoo
+                                                          metadata:metadataBar]);
```

Comment:
Another giant wall of text.  Can we improve with a helper of some kind?

Reply:
done